Amplify/Categories/DataStore/Model/Internal/Schema/ModelSchema+Definition.swift

```diff
@@ -119,6 +119,22 @@ public enum ModelFieldNullability {
     }
 }
 
+/// - Warning: Although this has `public` access, it is intended for internal & codegen use and should not be used
+///   directly by host applications. The behavior of this may change without warning.
+public enum ModelFieldWritability {
```

Comment:
Naming suggestions are more than welcome :)

Reply:
We've already got a "Nullability", this name seems reasonable. But is it needed? If we only have two cases, neither one with associated values, we're basically writing an enum to model a boolean. Why not just use a Bool?

Reply:
The only reason I opted for an enum is to be able to represent other values in the future (like `[writeOnly](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.9.4)`) without needed to change type. But as I'm not sure if/when/how/why we would need other values, I'm more than happy to switch to a `Bool` if we don't feel is useful

Reply:
If we've got a current use case in mind, feel free to keep it, otherwise, it seems like we're adding a bit of complexity we don't need and I'd vote to remove it. I'm not saying "writeOnly" isn't a valid use case, only that we don't have a current need for it.

If we're attempting to conform to JSONSchema or some other standard, we probably have more work to do than just adding an enum case :)